src/mds/MDSRank.cc

```diff
@@ -2526,15 +2524,12 @@ void MDSRankDispatcher::handle_asok_command(
     cmd_getval(g_ceph_context, cmdmap, "scrubops", scrubop_vec);
     cmd_getval(g_ceph_context, cmdmap, "path", path);
     cmd_getval(g_ceph_context, cmdmap, "tag", tag);
+
     /* if there are more than one mds active and a recursive scrub is requested,
      * dishonor the request
      */
-    unsigned active = mdsmap->get_num_mds(MDSMap::DaemonState::STATE_ACTIVE);
-    unsigned stopping = mdsmap->get_num_mds(MDSMap::DaemonState::STATE_STOPPING);
-    bool is_recursive = std::find(scrubop_vec.begin(), scrubop_vec.end(), string("recursive")) != scrubop_vec.end();
-    // if (mdsmap->get_max_mds() > 1 &&
-    if ((active + stopping > 1) && is_recursive) {
-      ss << "There are more than one MDS active. Hence a recursive scrub "
+    if (mdsmap->get_max_mds() > 1) {
```

Comment:
@batrick Don't we need to test for the `recursive` flag here ?

Reply:
No. Consider the directory may be fragmented and other fragments are on a different MDS.